File: backend/src/entities/person.go
package entities

import (
	"time"

	"github.com/brianvoe/gofakeit/v6"
)

type Person struct {
	ID           int64 `db:"person_id"`
	Name         string
	Email        string
	Phone        string
	Description  string
	Birthday     time.Time
	PersonSkills []PersonSkill `db:"person_skill"`
}

func NewRandomPerson(skills []Skill) Person {
	p := Person{
		ID:          gofakeit.Int64(),
		Name:        gofakeit.Name(),
		Email:       gofakeit.Email(),
		Phone:       gofakeit.Phone(),
		Description: gofakeit.Phrase(),
		Birthday:    gofakeit.DateRange(time.Now().AddDate(-60, 0, 0), time.Now().AddDate(-18, 0, 0)),
	}
	for _, s := range skills {
		p.PersonSkills = append(p.PersonSkills, PersonSkill{
			PersonID:      p.ID,
			SkillID:       s.ID,
			Value:         gofakeit.Number(0, 100),
			Progress:      gofakeit.Number(0, 100),
		})
	}
	return p
}

File: frontend/lib/entities/person.dart
import 'package:flutter/material.dart';
import 'package:frontend/entities/personSkill.dart';

class Person {
  final String personId;
  final String name;
  final String email;
  final String phone;
  final String description;
  final String birthday;
  List<PersonSkill> personSkills;

  Person(
      {@required this.personId,
      @required this.name,
      @required this.email,
      @required this.phone,
      @required this.birthday,
      @required this.description,
      this.personSkills});

  factory Person.fromJson(Map<String, dynamic> json) {
    List<PersonSkill> personSkills = [];
    if (json['PersonSkills'] != null) {
      personSkills = json['PersonSkills']
          .map<PersonSkill>((item) => PersonSkill.fromJson(item))
          .toList();
    }
    final data = Person(
        personId: json['PersonId'],
        name: json['Name'],
        email: json['Email'],
        phone: json['Phone'],
        description: json['Description'],
        birthday: json['Birthday'],
        personSkills: personSkills);
    return data;
  }
}
